packages/mercury-case-analysis/src/rating.ts

```diff
@@ -0,0 +1,163 @@
+import { generateObject, LanguageModel } from "mongodb-rag-core/aiSdk";
+import { extractZodSchemaDescriptions } from "./utils";
+import z from "zod";
+import { stripIndent } from "common-tags";
+import YAML from "yaml";
+
+const ratingSchema = z
+  .number()
+  .min(1)
+  .max(5)
+  .describe("A score from 1 (poor) to 5 (excellent).");
+
+const makeRationaleSchema = (name: string) =>
+  z
+    .string()
+    .optional()
+    .describe(
+      `A terse and direct chain of thought or rationale for why you gave the ${name} score.`
+    );
+
+export const promptResponseRatingSchema = z.object({
+  answer_fit: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("answer fit"),
+    })
+    .describe(
+      stripIndent`
+        How well the expected answer actually matches the prompt.
+        A high score indicates that the answer is directly relevant to the prompt and provides a clear and concise response.
+        A low score indicates that the answer is not directly relevant to the prompt or contains extraneous information.
+      `
+    ),
+  answer_reasonableness: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("answer reasonableness"),
+    })
+    .describe(
+      stripIndent`
+        How reasonable it would be to expect an LLM to produce the given response from the given prompt.
+        A high score indicates that the answer is reasonable and likely to be produced by an LLM.
+        A low score indicates that the answer is not reasonable and likely to be produced by an LLM.
+      `
+    ),
+  business_impact: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("business impact"),
+    })
+    .describe(
+      stripIndent`
+        The business impact/relevance of the question and answer.
+        High impact questions are likely to generate new revenue or prevent customer churn.
+        Examples of high impact topics: market positioning, technical support, product roadmap, etc.
+        Low impact questions are likely to be easily answered by a simple search and/or have little impact on the business.
+        Examples of low impact prompts: When was MongoDB founded?, What is the latest version of MongoDB?, etc.
+      `
+    ),
+  prompt_clarity: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("prompt clarity"),
+    })
+    .describe(
+      stripIndent`
+        How well-formulated and clear the prompt is.
+        A high score indicates that the prompt is clear and focused on a specific topic.
+        A low score indicates that the prompt is unclear or too broad.
```

Comment:
these are a lot of description for the feild. not usual pattern, but if it works, i guess fine..